Fix -Wint-conversion with gcc14 and musl, Eina_Thread is defined as uintptr_t

diff -rup src/lib/eina/eina_debug_bt.c.orig src/lib/eina/eina_debug_bt.c
--- a/src/lib/eina/eina_debug_bt.c
+++ b/src/lib/eina/eina_debug_bt.c
@@ -339,7 +339,7 @@ _trace_cb(void *data EINA_UNUSED)
       ((_eina_debug_thread_active_num) * sizeof(int));
    // now collect per thread
    for (i = 0; i < _eina_debug_thread_active_num; i++)
-      _collect_bt(_eina_debug_thread_active[i].thread);
+      _collect_bt((pthread_t)_eina_debug_thread_active[i].thread);
    // we're done probing. now collec all the "i'm done" msgs on the
    // semaphore for every thread + mainloop
    for (i = 0; i < (_eina_debug_thread_active_num); i++)
